src/Microsoft.AspNet.Diagnostics.Elm/ElmCaptureMiddleware.cs

```diff
@@ -48,11 +45,11 @@ public ElmCaptureMiddleware(RequestDelegate next, ILoggerFactory factory, IOptio
         /// Takes the info from the given HttpContext and copies it to an HttpInfo object
         /// </summary>
         /// <returns>The HttpInfo for the current elm context</returns>
-        private static HttpInfo GetHttpInfo(HttpContext context, Guid requestId)
+        private static HttpInfo GetHttpInfo(HttpContext context)
         {
             return new HttpInfo()
             {
-                RequestID = requestId,
+                RequestID = context.GetFeature<IHttpRequestIdentifierFeature>().TraceIdentifier,
```

Comment:
GetFeature can never assume anything about other components. Hosting may add it but other middleware can modify or remove it.